app/src/main/java/de/tum/in/tumcampusapp/utils/CacheManager.kt

```diff
@@ -0,0 +1,65 @@
+package de.tum.`in`.tumcampusapp.utils
+
+import android.content.Context
+import de.tum.`in`.tumcampusapp.api.tumonline.AccessTokenManager
+import de.tum.`in`.tumcampusapp.api.tumonline.CacheControl
+import de.tum.`in`.tumcampusapp.api.tumonline.TUMOnlineClient
+import de.tum.`in`.tumcampusapp.component.tumui.lectures.model.LecturesResponse
+import de.tum.`in`.tumcampusapp.component.ui.chat.ChatRoomController
+import okhttp3.Cache
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import java.io.IOException
+
+class CacheManager(private val context: Context) {
+
+    val cache: Cache
+        get() = Cache(context.cacheDir, 10 * 1024 * 1024) // 10 MB
+
+    fun fillCache() {
+        if (!AccessTokenManager.hasValidAccessToken(context)) {
+            return
+        }
+
+        syncCalendar()
+        syncPersonalLectures()
+    }
+
+    fun syncCalendar() {
+        try {
+            TUMOnlineClient
+                    .getInstance(context)
+                    .getCalendar(CacheControl.NO_CACHE)
+                    .execute()
+        } catch (e: IOException) {
+            Utils.log(e, "Error while loading calendar in CacheManager")
+        }
+    }
+
+    private fun syncPersonalLectures() {
+        TUMOnlineClient
+                .getInstance(context)
+                .getPersonalLectures(CacheControl.NO_CACHE)
```

Comment:
It his correct?
We call this in `fillCache()`, but then set `NO_CACHE`?

Ok, I still think this is confusing...
This should probably be called `BYPASS_CACHE`

Reply:
I changed the naming accordingly and changed the method to `CacheControl.USE_CACHE`, because we only want to refresh the content in the background if it’s past its `max-age` (same behavior as before the refactoring).